### PR TITLE
Fix mac builds

### LINKED ISSUE #23681: mac_debug tests are failing due to skipping tests and not being able to find results
The mac_debug tests in jenkins are failing constantly due to all of the tests being skipped and then not being able to find any results. 

One of the failures is linked here with an excerpt of the console log:
https://ci.dot.net/job/dotnet_roslyn/view/Official%20Builds/job/master/job/mac_debug/2113/
```csharp
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/BasicCodeStyleTests/netcoreapp2.0/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpCodeStyleTests/netcoreapp2.0/Microsoft.CodeAnalysis.CSharp.CodeStyle.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpCompilerSymbolTest/netcoreapp2.0/Roslyn.Compilers.CSharp.Symbol.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpCompilerSyntaxTest/netcoreapp2.0/Roslyn.Compilers.CSharp.Syntax.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpScriptingTest/netcoreapp2.0/Microsoft.CodeAnalysis.CSharp.Scripting.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CodeStyleTests/netcoreapp2.0/Microsoft.CodeAnalysis.CodeStyle.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/ScriptingTest/netcoreapp2.0/Microsoft.CodeAnalysis.Scripting.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/VBCSCompilerTests/netcoreapp2.0/Roslyn.Compilers.CompilerServer.UnitTests.dll
08:00:41 [xUnit] [INFO] - Starting to record.
08:00:41 [xUnit] [INFO] - Processing xUnit.Net-v2 (default)
08:00:41 [xUnit] [INFO] - [xUnit.Net-v2 (default)] - No test report file(s) were found with the pattern '**/xUnitResults/*.xml' relative to '/Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug' for the testing framework 'xUnit.Net-v2 (default)'.  Did you enter a pattern relative to the correct directory?  Did you generate the result report(s) for 'xUnit.Net-v2 (default)'?
08:00:41 
[xUnit] [ERROR] - No test reports found for the metric 'xUnit.Net' with the resolved pattern '**/xUnitResults/*.xml'. Configuration error?.```

### PR DESCRIPTION
**Pull request by jaredpar**
The Mac builds were primarily failing due to a regex check which behaves differently on Mac and Linux. In Linux an empty string matched all inputs while it does not on Mac builds. Added an explicit check for the empty case. 

Closes #23681

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @khyperia, @agocke for review

**Comment by jaredpar** [comment]
test mac please

**Comment by jaredpar** [comment]
Looks like this fixed up the Mac issue.

File: build/scripts/tests.sh [hunk 1]
```diff
@@ -45,8 +45,8 @@ do
     runtimeconfig_json="${file_name%.*}".runtimeconfig.json
 
     # If the user specifies a test on the command line, only run that one
-    # "${2:-}" => take second arg, empty string if unset (regex always matches if empty)
-    if [[ ! "${file_name}" =~ "${2:-}" ]]
+    # "${2:-}" => take second arg, empty string if unset
+    if [[ ("${2:-}" != "") && (! "${file_name}" =~ "${2:-}") ]]
```
Comment (agocke) [inline build/scripts/tests.sh]: This is a disturbing OS difference, but LGTM.

File: build.sh [hunk 2]
```diff
@@ -122,6 +122,13 @@ then
     build_args+=" /p:BootstrapBuildPath=${bootstrap_path}"
 fi
 
+# https://github.com/dotnet/roslyn/issues/23736
+UNAME="$(uname)"
+if [[ "$UNAME" == "Darwin" ]]
+then
+    build_args+=" /p:UseRoslynAnalyzers=false"
```
Comment (agocke) [inline build.sh]: Can you explain what's up with the analyzers?

Comment (jaredpar) [reply build.sh]: No idea. That's why I filed the bug to track this down 😄

File: build/scripts/tests.sh (hunk 1 above)
Comment (jaredpar) [reply build/scripts/tests.sh]: Agree. Tried to lookup docs on this (mostly to verify I wasn't interpreting this wrong). But experimentally at least it appears to be a difference between the OS.

**Comment by jcouv** [comment]
Thanks 👍

**Pull request merged**